### PR TITLE
fix: make --output - implies --no-tty

### LINKED ISSUE #1268: oras blob fetch failed to output to stdout in TTY mode
### What happened in your environment?

I tried to fetch a blob with `oras blob fetch` and output the content into stdout in terminal mode. The blob content is not printed to stdout as expected

```console
➜  oras blob fetch ghcr.io/oras-project/oras@sha256:488a49baab43328b45444fc9c13861da4408f279ba5247569bbe06a732265fdc --output -
✓ Downloaded  application/octet-stream                                                                  167/167  B 100.00%     0s
  └─ sha256:488a49baab43328b45444fc9c13861da4408f279ba5247569bbe06a732265fdc
```

With `--no-tty`, the content can be printed
```console
➜  oras blob fetch ghcr.io/oras-project/oras@sha256:488a49baab43328b45444fc9c13861da4408f279ba5247569bbe06a732265fdc --output - --no-tty
{"architecture":"unknown","os":"unknown","config":{},"rootfs":{"type":"layers","diff_ids":["sha256:31e4a5298aed7d277e1784bc57ff297a450e25eb59d5b7ac89f6bb229ac1a18a"]}}%
```

### What did you expect to happen?

The blob content should be printed to stdout

### How can we reproduce it?

Run below command
```bash
oras blob fetch ghcr.io/oras-project/oras@sha256:488a49baab43328b45444fc9c13861da4408f279ba5247569bbe06a732265fdc --output -
```

### What is the version of your ORAS CLI?

v1.2.0-beta.1

### What is your OS environment?

ubuntu 20.04

### Are you willing to submit PRs to fix it?

- [ ] Yes, I am willing to fix it.

### PR DESCRIPTION
**Pull request by wangxiaoxuan273**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #1268 

**Please check the following list**:
- [ ]  Does the affected code have corresponding tests, e.g. unit test, E2E test?
- [ ]  Does this change require a documentation update?
- [ ]  Does this introduce breaking changes that would require an announcement or bumping the major version?
- [ ]  Do all new files have an appropriate license header?

### CODE REVIEW

**Comment by wangxiaoxuan273** [comment]
As per [comment in the issue](https://github.com/oras-project/oras/issues/1268#issuecomment-1992008000), now `--output -` implies `--no-tty` in `oras blob fetch`, but we also allow user to explicitly put `--no-tty=false` with `--output -`.

Current display:
Screenshot 2024-03-13 at 19 00 22

File: cmd/oras/root/blob/fetch.go [hunk 1]
```diff
@@ -157,6 +161,11 @@ func (opts *fetchBlobOptions) doFetch(ctx context.Context, src oras.ReadOnlyTarg
 			}
 		}()
 		writer = file
+	} else {
```
Comment (TerryHowe) [inline cmd/oras/root/blob/fetch.go:164]: Seems like these changes would be avoided with a validation in CheckArgs

Comment (qweeah) [reply cmd/oras/root/blob/fetch.go:164]: @TerryHowe is right, this validation can be done after all options are parsed via resetting `common.TTY` when needed.

Comment (wangxiaoxuan273) [reply cmd/oras/root/blob/fetch.go]: updated

File: cmd/oras/internal/option/common.go [hunk 2]
```diff
@@ -25,20 +25,22 @@ import (
...
-	noTTY bool
+	NoTTY bool
 }
 
 // ApplyFlags applies flags to a command flag set.
 func (opts *Common) ApplyFlags(fs *pflag.FlagSet) {
 	fs.BoolVarP(&opts.Debug, "debug", "d", false, "output debug logs (implies --no-tty)")
 	fs.BoolVarP(&opts.Verbose, "verbose", "v", false, "verbose output")
-	fs.BoolVarP(&opts.noTTY, "no-tty", "", false, "[Preview] do not show progress output")
+	fs.BoolVarP(&opts.NoTTY, "no-tty", "", false, "[Preview] do not show progress output")
```
Comment (qweeah) [inline cmd/oras/internal/option/common.go:43]:
```suggestion
	fs.BoolVarP(&opts.NoTTY, NoTTYFlag, "", false, "[Preview] do not show progress output")
```

File: cmd/oras/internal/option/common_test.go [hunk 3]
```diff
@@ -28,3 +30,45 @@ func TestCommon_FlagsInit(t *testing.T) {
...
+	testTTY := &os.File{}
+	tests := []struct {
+		name        string
+		flagPresent bool
+		toSTDOUT    bool
+		noTTY       bool
+		expectedTTY *os.File
+	}{
+		{
+			"output path == -, --no-tty flag not used", false, true, false, nil,
```
Comment (qweeah) [inline cmd/oras/internal/option/common_test.go:44]:
```suggestion
			"output to STDOUT without setting --no-tty, reset TTY", false, true, false, nil,
```

File: cmd/oras/internal/option/common_test.go [hunk 4]
```diff
@@ -28,3 +30,45 @@ func TestCommon_FlagsInit(t *testing.T) {
...
+		},
+		{
+			"output path == -, --no-tty", true, true, true, nil,
```
Comment (qweeah) [inline cmd/oras/internal/option/common_test.go:47]:
```suggestion
			"--no-tty set to true, reset TTY", true, true, true, nil,
```

File: cmd/oras/internal/option/common_test.go [hunk 5]
```diff
@@ -28,3 +30,45 @@ func TestCommon_FlagsInit(t *testing.T) {
...
+		},
+		{
+			"output path == -, --no-tty=false", true, true, false, testTTY,
+		},
+		{
+			"output path != -, --no-tty flag not used", false, false, false, testTTY,
+		},
+		{
+			"output path != -, --no-tty", true, false, true, nil,
```
Comment (qweeah) [inline cmd/oras/internal/option/common_test.go:56]:
```suggestion
			"--no-tty set to true, reset TTY", true, false, true, nil,
```

File: cmd/oras/internal/option/common_test.go [hunk 6]
```diff
@@ -28,3 +30,45 @@ func TestCommon_FlagsInit(t *testing.T) {
...
+		expectedTTY *os.File
+	}{
+		{
+			"output path == -, --no-tty flag not used", false, true, false, nil,
+		},
+		{
+			"output path == -, --no-tty", true, true, true, nil,
+		},
+		{
+			"output path == -, --no-tty=false", true, true, false, testTTY,
```
Comment (qweeah) [inline cmd/oras/internal/option/common_test.go:50]:
```suggestion
			"output to STDOUT, --no-tty set to false", true, true, false, testTTY,
```

File: cmd/oras/internal/option/common_test.go [hunk 7]
```diff
@@ -28,3 +30,45 @@ func TestCommon_FlagsInit(t *testing.T) {
...
+		},
+		{
+			"output path != -, --no-tty flag not used", false, false, false, testTTY,
```
Comment (qweeah) [inline cmd/oras/internal/option/common_test.go:53]:
```suggestion
			"not output to STDOUT", false, false, false, testTTY,
```

**Review by qweeah — APPROVED** [review]
LGTM with minor suggestion: suggest to align unit test description with real e2e use case.

**Pull request merged**